src/artemis/experiment_plans/tests/conftest.py

```diff
@@ -191,3 +238,203 @@ def mock_rotation_subscriptions(test_rotation_params):
+
+
+def fake_read(obj, initial_positions, _):
+    initial_positions[obj] = 0
+    yield Msg("null", obj)
+
+
+def do_rotation_plan_for_tests(
+    run_engine,
+    callbacks,
+    sim_und,
+    sim_synch,
+    sim_slits,
+    sim_flux,
+    sim_att,
+    expt_params,
+    sim_sgon,
+    sim_zeb,
+    sim_bl,
+    sim_det,
+):
+    with (
+        patch(
+            "bluesky.preprocessors.__read_and_stash_a_motor",
+            fake_read,
+        ),
+        patch(
+            "artemis.experiment_plans.rotation_scan_plan.RotationCallbackCollection.from_params",
+            lambda _: callbacks,
+        ),
+        patch("dodal.beamlines.i03.undulator", lambda: sim_und),
+        patch("dodal.beamlines.i03.synchrotron", lambda: sim_synch),
+        patch("dodal.beamlines.i03.s4_slit_gaps", lambda: sim_slits),
+        patch("dodal.beamlines.i03.flux", lambda: sim_flux),
+        patch("dodal.beamlines.i03.attenuator", lambda: sim_att),
+    ):
+        run_engine(
+            rotation_scan_plan(
+                expt_params,
+                sim_sgon,
+                sim_zeb,
+                sim_bl,
+                sim_att,
+                sim_det,
+            )
+        )
+
+
+def setup_and_run_rotation_plan_for_tests(
+    RE: RunEngine,
+    test_params: RotationInternalParameters,
+    smargon: Smargon,
+    zebra: Zebra,
+    eiger: EigerDetector,
+    attenuator: Attenuator,
+    detector_motion: DetectorMotion,
+    backlight: Backlight,
+    mock_rotation_subscriptions: RotationCallbackCollection,
+    synchrotron: Synchrotron,
+    s4_slit_gaps: S4SlitGaps,
+    undulator: Undulator,
+    flux: Flux,
+):
+    from functools import partial
+
+    def side_set_w_return(obj, *args):
+        obj.sim_put(*args)
+        return DEFAULT
+
+    smargon.omega.velocity.set = MagicMock(
+        return_value=Status(done=True, success=True),
+        side_effect=partial(side_set_w_return, smargon.omega.velocity),
+    )
+    smargon.omega.set = MagicMock(
+        return_value=Status(done=True, success=True),
+        side_effect=partial(side_set_w_return, smargon.omega.user_readback),
+    )
+    smargon.x.set = MagicMock(
+        return_value=Status(done=True, success=True),
+        side_effect=partial(side_set_w_return, smargon.x.user_readback),
+    )
+    smargon.y.set = MagicMock(
+        return_value=Status(done=True, success=True),
+        side_effect=partial(side_set_w_return, smargon.y.user_readback),
+    )
+    smargon.z.set = MagicMock(
+        return_value=Status(done=True, success=True),
+        side_effect=partial(side_set_w_return, smargon.z.user_readback),
+    )
+    smargon.chi.set = MagicMock(
+        return_value=Status(done=True, success=True),
+        side_effect=partial(side_set_w_return, smargon.chi.user_readback),
+    )
+    smargon.phi.set = MagicMock(
+        return_value=Status(done=True, success=True),
+        side_effect=partial(side_set_w_return, smargon.phi.user_readback),
+    )
```

Comment:
Could: [Snap!](https://github.com/DiamondLightSource/python-artemis/pull/832/files#diff-4515d8d2bbf03eeac15b1f2d892eec01e2ed10de3eb06ed974ed9fa788f9c5d4) Ultimately we should get this into [ophyd](https://github.com/bluesky/ophyd/issues/1135) but in the meantime I think this actually belongs in the `smargon` fixture.

Reply:
I think we should also tidy up all the fixtures in general,
https://github.com/DiamondLightSource/python-artemis/issues/837